### DISCUSSION TITLE
ESPHome Installation fails

### DISCUSSION THREAD

**Discussion by bajangti** [Blackymas/NSPanel_HA_Blueprint#402] [Q&A] [answered]
When I try to install ESPHome I keep getting a preparation failed. When I try to Validate the config it is successful.
 
I am running ESPHome v2023.2.4 with HA 2023.3.1.

This is the error log:

 INFO Reading configuration /config/esphome/nspanelcorridor.yaml...
INFO Updating https://github.com/Blackymas/NSPanel_HA_Blueprint@main
WARNING GPIO4 is a Strapping PIN and should be avoided.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Generating C++ source...
INFO Compiling app...
Processing nspanelcorridor (board: esp32dev; framework: arduino; platform: platformio/espressif32 @ 5.2.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch3

warning: Calling missing SConscript without error is deprecated.
Transition by adding must_exist=False to SConscript calls.
Missing SConscript '/data/cache/platformio/packages/framework-arduinoespressif32/tools/platformio-build.py'
File "/data/cache/platformio/platforms/espressif32/builder/frameworks/arduino.py", line 41, in <module>
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Library Manager: Installing FS
Warning! Could not find the package with 'FS' requirements for your system 'linux_x86_64'
Library Manager: Installing Update
Warning! Could not find the package with 'Update' requirements for your system 'linux_x86_64'
Library Manager: Installing ESPmDNS
Warning! Could not find the package with 'ESPmDNS' requirements for your system 'linux_x86_64'
Library Manager: Installing WiFiClientSecure
Warning! Could not find the package with 'WiFiClientSecure' requirements for your system 'linux_x86_64'
Dependency Graph
|-- AsyncTCP-esphome @ 1.2.2
|-- WiFi @ 1.2.7
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- AsyncTCP-esphome @ 1.2.2
|-- DNSServer @ 1.1.0
|-- ArduinoJson @ 6.18.5
|-- HttpClient @ 2.2.0
Compiling /data/nspanelcorridor/.pioenvs/nspanelcorridor/src/esphome/components/adc/adc_sensor.cpp.o
Compiling /data/nspanelcorridor/.pioenvs/nspanelcorridor/src/esphome/components/api/api_connection.cpp.o
In file included from src/esphome/components/socket/socket.h:5,
                 from src/esphome/components/api/api_frame_helper.h:13,
                 from src/esphome/components/api/api_connection.h:3,
                 from src/esphome/components/api/api_connection.cpp:1:
src/esphome/components/socket/headers.h:116:10: fatal error: sys/ioctl.h: No such file or directory
 #include <sys/ioctl.h>
          ^~~~~~~~~~~~~
compilation terminated.
*** [/data/nspanelcorridor/.pioenvs/nspanelcorridor/src/esphome/components/api/api_connection.cpp.o] Error 1
In file included from src/esphome/components/adc/adc_sensor.h:4,
                 from src/esphome/components/adc/adc_sensor.cpp:1:
src/esphome/core/hal.h:14:10: fatal error: esp_attr.h: No such file or directory

******************************************************************
* Looking for esp_attr.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:esp_attr.h"
* Web  > https://registry.platformio.org/search?q=header:esp_attr.h
*
******************************************************************

 #include <esp_attr.h>
          ^~~~~~~~~~~~
compilation terminated.
*** [/data/nspanelcorridor/.pioenvs/nspanelcorridor/src/esphome/components/adc/adc_sensor.cpp.o] Error 1
========================== [FAILED] Took 3.40 seconds ==========================

**Comment by edwardtfn** [comment]
Are you still having this issue?

**Reply by bajangti** [reply]
No, I resolved it by reinstalling ESP Home like @denes44 .

**Comment by deejaybeam** [comment]
looks like related to: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/249, https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/121, https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/172

**Comment by denes44** [comment, marked as answer]
I had the same issue a couple days before when I started using this project.
Reinstalling the esphome HA addon worked for me. It does not change your files in the `config/esphome` directory, so it can be reinstalled quickly (however always make a backup before doing thing like this :) )

**Reply by VonSquiggles** [reply]
I had the same issue and this also fixed it for me.

**Reply by electronicadventures** [reply]
reinstall, restart. error now gone

**Reply by Davis8483** [reply]
Reinstalling did not resolve the issue for me. It also appears to be specific to compiling for the esp32. The esp8266 compiles just fine.

**Reply by edwardtfn** [reply]
There's no esp8266 on this project... 🤪
If you are having an issue, do you mind reporting in the issues area with logs and yaml?

**Reply by Davis8483** [reply]
oh, sorry. I found this thread on Google when I was researching compilation errors with ESP Home. Compiling anything seems to work finally, after reinstalling ESP Home about 5 times now.
